### ISSUE TITLE
LoRA adapter not applied in Remix mode; Custom mode hangs with LoRA loaded on xl-base

### ISSUE THREAD

**Issue by hs86236** [ace-step/ACE-Step-1.5#1112]
A trained PEFT LoRA adapter is loaded successfully and demonstrably affects output in text2music generation, but produces no audible effect when used in Remix mode, and causes Custom mode to hang indefinitely during inference.

Environment

ACE-Step commit: 82252c2 (current master)
GPU: NVIDIA RTX PRO 6000 Blackwell Server Edition (sm_120)
PyTorch: 2.10.0+cu128
CUDA: 13.0 driver, compute capability 12.0
OS: Linux (RunPod container)
Python: 3.11 (venv)
Model: acestep-v15-xl-base + acestep-5Hz-lm-4B
Flash Attention: enabled
Torch Compile: disabled

LoRA training details

Adapter type: PEFT LoRA
Rank: 64
Alpha: 128
Dropout: 0.1
Target modules: attention layers (default)
Training mode: fixed (corrected timesteps + CFG dropout)
Dataset: 17 full-mix audio tracks, ~30 minutes total
Epochs: 1000 
Training completed successfully with clean convergence pattern

To Reproduce

Initialize acestep-v15-xl-base + acestep-5Hz-lm-4B
Load trained LoRA adapter via UI (confirmation received: "LoRA loaded")
Set LoRA scale to 0.5

Remix mode reproduction:

Switch to Remix mode
Upload source audio (48kHz stereo WAV, 148 seconds)
Set Remix Strength: 0.3, Cover Strength: 0.7
Provide music caption and lyrics
Generate
Toggle LoRA off (or change scale to 0.0)
Generate again with identical settings
Result: outputs are audibly identical — LoRA has no effect

Custom mode reproduction:

Switch to Custom mode
Same caption, lyrics set to [Instrumental]
Generate
Result: inference hangs at "Preparing inputs..." with no further log output; Python process eventually becomes idle (0% CPU) with no completion
Expected behavior

LoRA scale changes should produce audible differences in output
LoRA should apply consistently across Remix, Custom, and Text2music modes
Custom mode should complete generation

Actual behavior

Text2music mode: LoRA applies correctly — audible difference between scale 0.0 and scale 0.5+ ✅
Remix mode: LoRA appears to be bypassed — identical output with/without LoRA loaded ❌
Custom mode: Generation hangs indefinitely after "Preparing inputs..." log line ❌

Terminal log (Custom mode hang)
2026-04-17 22:58:49.620 | INFO | acestep.inference:generate_music:604 - [generate_music] Final inputs: dit_input_caption='[...]', dit_input_lyrics='[Instrumental]'
2026-04-17 22:58:49.631 | INFO | acestep.core.generation.handler.generate_music:generate_music:278 - [generate_music] Starting generation...
2026-04-17 22:58:49.631 | INFO | acestep.core.generation.handler.generate_music:generate_music:281 - [generate_music] Preparing inputs...
[no further output; process idle]

Related observations

Earlier attempts at Cover mode produced CUBLAS_STATUS_NOT_INITIALIZED errors originating in vector_quantize_pytorch/residual_fsq.py during FSQ tokenizer forward pass
Clearing __pycache__ and restarting Gradio did not resolve either issue
The same LoRA file works correctly when loaded into text2music mode

Possibly related

Issue #728 (LoRA effect not noticeable)
Issue #434 (LoRA garbled after relaunch)
PR #701 (garbled audio after Remix/Repaint to Custom mode switch) — fix may not cover this specific path
ComfyUI issue #12638 (LoKr silently not applied)

**Comment by hs86236** [comment]
<h2 class="text-text-100 mt-3 -mb-1 text-[1.125rem] font-bold">Update: reproduced on A100 — this is NOT a Blackwell/sm_120 issue</h2>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">I migrated my entire setup to a different GPU to rule out hardware compatibility issues. <strong>The LoRA bypass in Remix and Cover modes reproduces identically on A100.</strong> This confirms the bug is in ACE-Step's code path, not in GPU-specific libraries (Flash Attention, CUBLAS, etc.).</p>
<h3 class="text-text-100 mt-2 -mb-1 text-base font-bold">Second test environment</h3>
<ul class="[li_&amp;]:mb-0 [li_&amp;]:mt-1 [li_&amp;]:gap-1 [&amp;:not(:last-child)_ul]:pb-1 [&amp;:not(:last-child)_ol]:pb-1 list-disc flex flex-col gap-1 pl-8 mb-3">
<li class="whitespace-normal break-words pl-2"><strong>ACE-Step commit</strong>: <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">82252c2</code> (current master, same as original report)</li>
<li class="whitespace-normal break-words pl-2"><strong>GPU</strong>: NVIDIA A100 80GB PCIe (sm_80) — ACE-Step's reference benchmark GPU</li>
<li class="whitespace-normal break-words pl-2"><strong>PyTorch</strong>: 2.10.0+cu128</li>
<li class="whitespace-normal break-words pl-2"><strong>CUDA</strong>: 12.4 driver</li>
<li class="whitespace-normal break-words pl-2"><strong>OS</strong>: Linux (Ubuntu 22.04, RunPod container)</li>
<li class="whitespace-normal break-words pl-2"><strong>Python</strong>: 3.11.15 (venv via uv)</li>
<li class="whitespace-normal break-words pl-2"><strong>Model</strong>: <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">acestep-v15-xl-base</code> + <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">acestep-5Hz-lm-4B</code></li>
<li class="whitespace-normal break-words pl-2"><strong>Flash Attention</strong>: enabled</li>
<li class="whitespace-normal break-words pl-2"><strong>Configuration Tier</strong>: unlimited (ACE-Step's best tier per <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">gpu_config.py</code>)</li>
<li class="whitespace-normal break-words pl-2"><strong>CPU offload</strong>: disabled (79GB VRAM available)</li>
</ul>
<h3 class="text-text-100 mt-2 -mb-1 text-base font-bold">Same LoRA, same reproduction steps</h3>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">Used the identical Capitol_Mix LoRA from the original report (PEFT, r=64, α=128, 1000 epochs trained in <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">fixed</code> mode, epoch 800 loss 0.2041). Same source audio (<code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">Too Marvelous For Words.wav</code>, 44.1kHz stereo, 2:29).</p>
<h3 class="text-text-100 mt-2 -mb-1 text-base font-bold">Results on A100 (identical to Blackwell)</h3>
<div class="overflow-x-auto w-full px-2 mb-6">
Mode | Behavior
-- | --
Text2music + LoRA | ✅ Works — audible LoRA effect, Capitol character comes through
Remix + LoRA | ❌ Same output with LoRA scale 0.5 vs 0.0 — adapter not applied
Cover + LoRA | ❌ Same output with LoRA loaded vs unloaded — adapter not applied

</div>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">Tested across multiple slider configurations:</p>
<ul class="[li_&amp;]:mb-0 [li_&amp;]:mt-1 [li_&amp;]:gap-1 [&amp;:not(:last-child)_ul]:pb-1 [&amp;:not(:last-child)_ol]:pb-1 list-disc flex flex-col gap-1 pl-8 mb-3">
<li class="whitespace-normal break-words pl-2">Remix Strength 0.9, Cover Strength 0.9 (maximum source preservation)</li>
<li class="whitespace-normal break-words pl-2">Remix Strength 0.3, Cover Strength 0.7 (bigger transformation)</li>
<li class="whitespace-normal break-words pl-2">LoRA Scale 0.3, 0.5, 0.7, 1.0</li>
</ul>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">None of these produce different output with vs without the LoRA loaded in Remix/Cover modes. Text2music mode produces clearly different output at the same scale values.</p>
<h3 class="text-text-100 mt-2 -mb-1 text-base font-bold">Cross-platform confirmation</h3>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">This bug now has confirmed reproduction on:</p>
<ul class="[li_&amp;]:mb-0 [li_&amp;]:mt-1 [li_&amp;]:gap-1 [&amp;:not(:last-child)_ul]:pb-1 [&amp;:not(:last-child)_ol]:pb-1 list-disc flex flex-col gap-1 pl-8 mb-3">
<li class="whitespace-normal break-words pl-2">NVIDIA RTX PRO 6000 Blackwell (sm_120) — original report</li>
<li class="whitespace-normal break-words pl-2">NVIDIA A100 80GB PCIe (sm_80) — this follow-up</li>
</ul>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">Both on commit <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">82252c2</code>, with same PyTorch/CUDA toolchain, same Capitol LoRA, same inputs. The only consistent factor is that <strong>the Remix/Cover code path does not apply loaded LoRA adapters, while the Text2music path does.</strong></p>
<h3 class="text-text-100 mt-2 -mb-1 text-base font-bold">Additional context for devs</h3>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">Noting for triage: the recent cover/repaint fixes (<code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">2e786dc</code>, <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">0779381</code>, <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">44b29c6</code>, <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">67c43bf</code>) and PR #701 address related issues but don't appear to resolve this specific LoRA application bypass. The LoRA loads successfully (UI confirms "LoRA loaded"), scales correctly in Text2music, but is somehow bypassed when the generation path switches to Remix/Cover.</p>
<p class="font-claude-response-body break-words whitespace-normal leading-[1.7]">Happy to help test any fixes or provide additional diagnostic output.</p>
